ui/components/MesheryPatterns.js

```diff
@@ -760,6 +770,52 @@ function MesheryPatterns({
     });
   };
 
+  const handlePublish = (formData) => {
+    const compatibilityStore = _.uniqBy(meshModels, (model) => _.toLower(model.displayName))
+      ?.filter((model) =>
+        formData?.compatibility?.some((comp) => _.toLower(comp) === _.toLower(model.displayName)),
+      )
+      ?.map((model) => model.name);
+
+    const payload = {
+      id: publishModal.pattern?.id,
+      catalog_data: {
+        ...formData,
+        compatibility: compatibilityStore,
+        type: _.toLower(formData?.type),
+      },
+    };
+
+    updateProgress({ showProgress: true });
+
+    publishPattern({
+      patternPayload: JSON.stringify(payload),
+    })
+      .unwrap()
+      .then(() => {
+        updateProgress({ showProgress: false });
+        if (user.role_names.includes('admin')) {
```

Comment:
please add a screenshot of what notification you are getting while publishing a design since i guess you don't have admin access